interfaces/machine_parameters/ring_section.py

```diff
@@ -227,31 +227,33 @@ def __init__(self, length, alpha_0,
             alpha_n[1] = alpha_1
             alpha_order_max = 1
 
-        for argument in kwargs:
+        for argument in list(kwargs.keys()):
             if 'alpha' in argument:
                 try:
                     order = int(argument.split('_')[-1])
-                    alpha_order_max = np.max([alpha_order_max, order])
-                    alpha_n[order] = kwargs[argument]
                 except ValueError:
                     raise excpt.InputError(
                         'The keyword argument ' + argument + ' was ' +
                         'interpreted as non-linear momentum compaction ' +
                         'factor. ' +
                         'The correct syntax is alpha_n.')
 
+                alpha_order_max = np.max([alpha_order_max, order])
+                alpha_n[order] = kwargs.pop(argument)
```

Comment:
Wouldn't make much difference, but you could avoid the repeated `np.max` call with by waiting until the end of the `for` loop and doing:
```python
alpha_order_max = np.max(list(alpha_n.keys()))
```

Reply:
Almost included

I have to catch the case where no alpha_n is declared. In that case I'd go for an if/else since the majority of the time no alpha_n is declared

```python
        if len(list(alpha_n.keys())) > 0:
            alpha_order_max = np.max(list(alpha_n.keys()))
        else:
            alpha_order_max = 0
```

would you still prefer a try/except?

Reply:
Ah, you don't include `alpha_0` as part of the `alpha_n` dictionary.  In that case I'd do:
```python
alpha_order_max = np.max(list(alpha_n.keys()) + [0])
```

Reply:
Included!

Reply:
Side note 2:  You can call `len` on a `dict`, it tells you the number of elements it contains.  So you will get identical result for `len(alpha_n)` as for `len(list(alpha_n.keys()))`.